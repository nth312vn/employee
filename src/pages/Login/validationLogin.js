import { loginField } from "../../constants/login";

export const validationLogin = () => {
  const { user, password } = loginField;
  return {
    [user]: (newValue) => {
      if (!newValue) {
        return "This field is required";
      }
    },
    [password]: (newValue) => {
      if (!newValue) {
        return "This field is required";
      }
    },
  };
};
