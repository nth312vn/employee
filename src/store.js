import { applyMiddleware, combineReducers, createStore } from "redux";
import thunk from "redux-thunk";
import { questionsReducer } from "./reducers/questions";
import { usersReducer } from "./reducers/users";

const rootReducer = combineReducers({
  users: usersReducer,
  questions: questionsReducer,
});
const store = createStore(rootReducer, applyMiddleware(thunk));
export default store;
