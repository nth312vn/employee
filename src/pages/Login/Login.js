import React from "react";
import {
  Button,
  Card,
  CardBody,
  Col,
  Container,
  Form,
  FormGroup,
  Input,
  Label,
  Row,
} from "reactstrap";
import { loginField } from "../../constants/login";
import useForm from "../../hooks/useForm";
import { validationLogin } from "./validationLogin";

const Login = () => {
  const { user, password } = loginField;
  const form = useForm(
    {
      [user]: "",
      [password]: "",
    },
    validationLogin()
  );
  const { value, error, handleChange, clearFrom, setField, handleSubmit } =
    form;
  const onSubmit = (formValues) => {
    console.log(formValues);
  };
  return (
    <Container style={{ marginTop: "50px" }}>
      <Row>
        <Label
          style={{ textAlign: "center", fontSize: "30px" }}
          className="pb-2 mr-sm-2 mb-sm-0"
        >
          Employee Polls Web App
        </Label>
      </Row>
      <Row>
        <Label
          style={{ textAlign: "center", fontSize: "30px", marginTop: "50px" }}
          className="pb-2 mr-sm-2 mb-sm-0"
        >
          Login
        </Label>
      </Row>
      <Row>
        <Col>
          <Card>
            <CardBody>
              <Form onSubmit={(e) => handleSubmit(e, onSubmit)}>
                <FormGroup className="pb-2 mr-sm-2 mb-sm-0">
                  <Label for="exampleEmail" className="mr-sm-2">
                    User
                  </Label>
                  <Input
                    type="text"
                    value={value[user]}
                    name={user}
                    id="exampleEmail"
                    placeholder="User"
                    onChange={handleChange}
                  />
                </FormGroup>
                <FormGroup className="pb-2 mr-sm-2 mb-sm-0">
                  <Label for="examplePassword" className="mr-sm-2">
                    Password
                  </Label>
                  <Input
                    type="password"
                    value={value[password]}
                    name={password}
                    id="examplePassword"
                    placeholder="Password"
                    onChange={handleChange}
                  />
                </FormGroup>
                <Button
                  style={{
                    marginTop: "30px",
                  }}
                  type="submit"
                  color="primary"
                >
                  Login
                </Button>
              </Form>
            </CardBody>
          </Card>
        </Col>
      </Row>
    </Container>
  );
};

export default Login;
