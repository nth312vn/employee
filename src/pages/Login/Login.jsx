import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { useNavigate } from "react-router-dom";
import {
  Button,
  Card,
  CardBody,
  Col,
  Container,
  Form,
  FormGroup,
  Input,
  Label,
  Row,
} from "reactstrap";
import { saveUser, setLastPathName } from "../../actions/usersAction";
import { loginField } from "../../constants/login";
import useForm from "../../hooks/useForm";
import { validationLogin } from "./validationLogin";

const Login = () => {
  const { user, password } = loginField;

  const form = useForm(
    {
      [user]: "",
      [password]: "",
    },
    validationLogin()
  );
  const users = useSelector((state) => state.users);
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const { value, error, handleChange, clearFrom, setError, handleSubmit } =
    form;
  const onSubmit = (formValues) => {
    clearFrom();
    if (users.listUser[formValues[user]]?.password === formValues[password]) {
      users.lastPathName ? navigate(users.lastPathName) : navigate("/");
      dispatch(setLastPathName(""));
      dispatch(saveUser(users.listUser[formValues[user]]));
      return;
    }
    setError("user name or password is invalid");
  };
  return (
    <Container style={{ marginTop: "50px" }}>
      <Row>
        <Label
          style={{ textAlign: "center", fontSize: "30px" }}
          className="pb-2 mr-sm-2 mb-sm-0"
        >
          Employee Polls Web App
        </Label>
      </Row>
      <Row>
        <Label
          style={{ textAlign: "center", fontSize: "30px", marginTop: "50px" }}
          className="pb-2 mr-sm-2 mb-sm-0"
        >
          <div data-testid="login-heading"> Login</div>
        </Label>
      </Row>
      <Row>
        <Col>
          <Card>
            <CardBody>
              <Form onSubmit={(e) => handleSubmit(e, onSubmit)}>
                <FormGroup className="pb-2 mr-sm-2 mb-sm-0">
                  <Label for="user" className="mr-sm-2">
                    User
                  </Label>
                  <Input
                    data-testid="username"
                    type="text"
                    value={value[user]}
                    name={user}
                    id="user"
                    placeholder="User"
                    onChange={handleChange}
                  />
                </FormGroup>
                <FormGroup className="pb-2 mr-sm-2 mb-sm-0">
                  <Label for="password" className="mr-sm-2">
                    Password
                  </Label>
                  <Input
                    data-testid="password"
                    type="password"
                    value={value[password]}
                    name={password}
                    id="password"
                    placeholder="Password"
                    onChange={handleChange}
                  />
                </FormGroup>
                {error && <div style={{ color: "red" }}>{error}</div>}
                <Button
                  style={{
                    marginTop: "30px",
                  }}
                  type="submit"
                  color="primary"
                >
                  Login
                </Button>
              </Form>
            </CardBody>
          </Card>
        </Col>
      </Row>
    </Container>
  );
};

export default Login;
