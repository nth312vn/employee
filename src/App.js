import { Route, Routes } from "react-router-dom";
import "./App.css";
import MainLayout from "./components/mainLayout/MainLayout";

import Login from "./pages/Login/Login";

function App() {
  return (
    <Routes>
      <Route path="/login" element={<Login />} />
      <Route path="/*" element={<MainLayout />} />
    </Routes>
  );
}

export default App;
