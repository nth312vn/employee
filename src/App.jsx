import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Route, Routes, useNavigate } from "react-router-dom";
import { handleInit } from "./actions/handleInitAction";
import "./App.css";
import MainLayout from "./components/mainLayout/MainLayout";

import Login from "./pages/Login/Login";

function App() {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const store = useSelector((state) => state);
  useEffect(() => {
    dispatch(handleInit());
    const user = localStorage.getItem("user");
    if (!user) {
      navigate("/login");
    }
  }, []);
  console.log(store);
  return (
    <Routes>
      <Route path="/login" element={<Login />} />
      <Route path="/*" element={<MainLayout />} />
    </Routes>
  );
}

export default App;
