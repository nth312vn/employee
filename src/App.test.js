/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable testing-library/render-result-naming-convention */
import { fireEvent, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import App from "./App";
import { _saveQuestion, _saveQuestionAnswer } from "./data/_DATA";
import Login from "./pages/Login/Login";
import NotFound from "./pages/NotFound/NotFound";
import store from "./store";

test("login", () => {
  const component = render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>
  );
  const heading = component.getByTestId("login-heading");
  expect(heading).toBeInTheDocument();
});
test("should return true for savequestion", async () => {
  const response = await _saveQuestion({
    optionOneText: "hihi",
    optionTwoText: "tesr",
    author: "tylermcginnis",
  }).catch((e) => e);
  expect(response).toBeTruthy();
});
test("should return true for correct parameters", async () => {
  const response = await _saveQuestionAnswer({
    authedUser: "sarahedo",
    qid: "8xf0y6ziyjabvozdd253nd",
    answer: "optionOne",
  });

  expect(response).toBeTruthy();
});

test("should return error for false parameters", async () => {
  const response = await _saveQuestionAnswer({
    authedUser: "sarahedo",
    qid: undefined,
    answer: "optionOne",
  }).catch((e) => e);

  expect(response).toBe("Please provide authedUser, qid, and answer");
});
test("should return error for savequestion", async () => {
  const response = await _saveQuestion({
    optionOneText: "hihi",
    author: "sarahedo",
  }).catch((e) => e);

  expect(response).toBe(
    "Please provide optionOneText, optionTwoText, and author"
  );
});
it("should render the component", () => {
  const component = render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>
  );
  expect(component).toBeDefined();
  expect(component).toMatchSnapshot();
});
it("should render the component login", () => {
  const component = render(
    <Provider store={store}>
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    </Provider>
  );
  expect(component).toBeDefined();
  expect(component).toMatchSnapshot();
});
it("should render the component notfound", () => {
  const component = render(
    <Provider store={store}>
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>
    </Provider>
  );
  expect(component).toBeDefined();
  expect(component).toMatchSnapshot();
});
it("should change username", async () => {
  const wrapper = render(
    <Provider store={store}>
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    </Provider>
  );
  const usernameInputElement = wrapper.getByTestId("username");
  fireEvent.change(usernameInputElement, { target: { value: "sarahedo" } });
  expect(usernameInputElement.value).toBe("sarahedo");
});
it("should change password", async () => {
  const wrapper = render(
    <Provider store={store}>
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    </Provider>
  );
  const usernameInputElement = wrapper.getByTestId("password");
  fireEvent.change(usernameInputElement, {
    target: { value: "wrongpassword" },
  });
  expect(usernameInputElement.value).toBe("wrongpassword");
});
