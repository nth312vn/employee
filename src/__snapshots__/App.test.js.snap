// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
        style="margin-top: 50px;"
      >
        <div
          class="row"
        >
          <label
            class="pb-2 mr-sm-2 mb-sm-0 form-label"
            style="text-align: center; font-size: 30px;"
          >
            Employee Polls Web App
          </label>
        </div>
        <div
          class="row"
        >
          <label
            class="pb-2 mr-sm-2 mb-sm-0 form-label"
            style="text-align: center; font-size: 30px; margin-top: 50px;"
          >
            <div
              data-testid="login-heading"
            >
               Login
            </div>
          </label>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="card"
            >
              <div
                class="card-body"
              >
                <form
                  class=""
                >
                  <div
                    class="pb-2 mr-sm-2 mb-sm-0 mb-3"
                  >
                    <label
                      class="mr-sm-2 form-label"
                      for="user"
                    >
                      User
                    </label>
                    <input
                      class="form-control"
                      data-testid="username"
                      id="user"
                      name="user"
                      placeholder="User"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="pb-2 mr-sm-2 mb-sm-0 mb-3"
                  >
                    <label
                      class="mr-sm-2 form-label"
                      for="password"
                    >
                      Password
                    </label>
                    <input
                      class="form-control"
                      data-testid="password"
                      id="password"
                      name="password"
                      placeholder="Password"
                      type="password"
                      value=""
                    />
                  </div>
                  <button
                    class="btn btn-primary"
                    style="margin-top: 30px;"
                    type="submit"
                  >
                    Login
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      style="margin-top: 50px;"
    >
      <div
        class="row"
      >
        <label
          class="pb-2 mr-sm-2 mb-sm-0 form-label"
          style="text-align: center; font-size: 30px;"
        >
          Employee Polls Web App
        </label>
      </div>
      <div
        class="row"
      >
        <label
          class="pb-2 mr-sm-2 mb-sm-0 form-label"
          style="text-align: center; font-size: 30px; margin-top: 50px;"
        >
          <div
            data-testid="login-heading"
          >
             Login
          </div>
        </label>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="card"
          >
            <div
              class="card-body"
            >
              <form
                class=""
              >
                <div
                  class="pb-2 mr-sm-2 mb-sm-0 mb-3"
                >
                  <label
                    class="mr-sm-2 form-label"
                    for="user"
                  >
                    User
                  </label>
                  <input
                    class="form-control"
                    data-testid="username"
                    id="user"
                    name="user"
                    placeholder="User"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="pb-2 mr-sm-2 mb-sm-0 mb-3"
                >
                  <label
                    class="mr-sm-2 form-label"
                    for="password"
                  >
                    Password
                  </label>
                  <input
                    class="form-control"
                    data-testid="password"
                    id="password"
                    name="password"
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
                <button
                  class="btn btn-primary"
                  style="margin-top: 30px;"
                  type="submit"
                >
                  Login
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render the component login 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
        style="margin-top: 50px;"
      >
        <div
          class="row"
        >
          <label
            class="pb-2 mr-sm-2 mb-sm-0 form-label"
            style="text-align: center; font-size: 30px;"
          >
            Employee Polls Web App
          </label>
        </div>
        <div
          class="row"
        >
          <label
            class="pb-2 mr-sm-2 mb-sm-0 form-label"
            style="text-align: center; font-size: 30px; margin-top: 50px;"
          >
            <div
              data-testid="login-heading"
            >
               Login
            </div>
          </label>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="card"
            >
              <div
                class="card-body"
              >
                <form
                  class=""
                >
                  <div
                    class="pb-2 mr-sm-2 mb-sm-0 mb-3"
                  >
                    <label
                      class="mr-sm-2 form-label"
                      for="user"
                    >
                      User
                    </label>
                    <input
                      class="form-control"
                      data-testid="username"
                      id="user"
                      name="user"
                      placeholder="User"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="pb-2 mr-sm-2 mb-sm-0 mb-3"
                  >
                    <label
                      class="mr-sm-2 form-label"
                      for="password"
                    >
                      Password
                    </label>
                    <input
                      class="form-control"
                      data-testid="password"
                      id="password"
                      name="password"
                      placeholder="Password"
                      type="password"
                      value=""
                    />
                  </div>
                  <button
                    class="btn btn-primary"
                    style="margin-top: 30px;"
                    type="submit"
                  >
                    Login
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      style="margin-top: 50px;"
    >
      <div
        class="row"
      >
        <label
          class="pb-2 mr-sm-2 mb-sm-0 form-label"
          style="text-align: center; font-size: 30px;"
        >
          Employee Polls Web App
        </label>
      </div>
      <div
        class="row"
      >
        <label
          class="pb-2 mr-sm-2 mb-sm-0 form-label"
          style="text-align: center; font-size: 30px; margin-top: 50px;"
        >
          <div
            data-testid="login-heading"
          >
             Login
          </div>
        </label>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="card"
          >
            <div
              class="card-body"
            >
              <form
                class=""
              >
                <div
                  class="pb-2 mr-sm-2 mb-sm-0 mb-3"
                >
                  <label
                    class="mr-sm-2 form-label"
                    for="user"
                  >
                    User
                  </label>
                  <input
                    class="form-control"
                    data-testid="username"
                    id="user"
                    name="user"
                    placeholder="User"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="pb-2 mr-sm-2 mb-sm-0 mb-3"
                >
                  <label
                    class="mr-sm-2 form-label"
                    for="password"
                  >
                    Password
                  </label>
                  <input
                    class="form-control"
                    data-testid="password"
                    id="password"
                    name="password"
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
                <button
                  class="btn btn-primary"
                  style="margin-top: 30px;"
                  type="submit"
                >
                  Login
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render the component notfound 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="not-found"
      >
        <img
          alt="not-found"
          src="https://www.pngitem.com/pimgs/m/561-5616833_image-not-found-png-not-found-404-png.png"
        />
        <a
          class="link-home"
          href="/"
        >
          Go Home
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="not-found"
    >
      <img
        alt="not-found"
        src="https://www.pngitem.com/pimgs/m/561-5616833_image-not-found-png-not-found-404-png.png"
      />
      <a
        class="link-home"
        href="/"
      >
        Go Home
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
